File: src/main/kotlin/leetCode/medium/RemovingStarsFromString.kt
package leetCode.medium

import org.junit.jupiter.api.Assertions.assertEquals

class RemovingStarsFromString {
    fun removeStars(s: String): String {
        val sb = StringBuilder()

        for (ch in s) {
            when (ch) {
                '*' -> sb.deleteCharAt(sb.length - 1)
                else -> sb.append(ch)
            }
        }

        return sb.toString()
    }
}

fun main(args: Array<String>) {
    val solution = RemovingStarsFromString()
    val example1 = solution.removeStars("leet**cod*e")
    assertEquals("lecoe", example1)

    val example2 = solution.removeStars("erase*****")
    assertEquals("", example2)
}

File: go/RemovingStarsFromString.go
package main

import "fmt"

func removeStars(s string) string {
	var sb []rune

	for _, ch := range s {
		switch ch {
		case '*':
			sb = sb[:len(sb)-1]
		default:
			sb = append(sb, ch)
		}
	}

	return string(sb)
}

func main() {
	fmt.Printf("example1 %s", removeStars("leet**cod*e"))
}
